metrics/metrics.go

```diff
@@ -27,3 +29,5 @@ func init() {
 	c.WithValues(version.Version, version.Revision).Inc()
 	goMetrics.Register(ns)
 }
+
+const ShimStatsRequestTimeout = 2 * time.Second
```

Comment:
Well, not a fan of fixed and short timeout because it depends on the load on the node. 

https://github.com/containerd/containerd/blob/ffddd4446c93caf0fc7029d2bd0a1b155641036b/runtime/v2/shim.go#L47

I think we can make it configurable like this.

Reply:
maybe use the context timeout value from the inbound stats request

Reply:
Thanks. I've added the configurable timeout.

Verified by:

- Running a shim that sleeps 5 seconds before responding to `Stats`.
- When running without timeout setting (default to 2 seconds), metrics of that shim will be missing
- When running with this timeout setting (10s). Metrics of that shim is present.

```
[timeouts]
    "io.containerd.timeout.metrics.shimstats" = "10s"
```